src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -319,24 +314,21 @@ internal static ProcessInfo[] GetProcessInfos(int? processIdFilter = null)
                                     throw new InvalidOperationException(SR.CouldntGetProcessInfos, new Win32Exception((int)status));
                                 }
 
+                                Debug.Assert(actualSize > 0 && actualSize <= bufferSize, $"Actual size reported by NtQuerySystemInformation was {actualSize} for a buffer of size={bufferSize}.");
+                                MostRecentSize = GetNewBufferSize(bufferSize, (int)actualSize);
                                 // Parse the data block to get process information
-                                processInfos = GetProcessInfos(MemoryMarshal.AsBytes<long>(buffer), processIdFilter);
-                                break;
+                                return GetProcessInfos(buffer.AsSpan(firstAlignedElementIndex), processIdFilter);
```

Comment:
```suggestion
                                return GetProcessInfos(buffer.AsSpan(firstAlignedElementIndex, (int)actualSize), processIdFilter);
```
to guarantee that GetProcessInfos does not touch uninitialized memory.